### DISCUSSION TITLE
如何使用其他程序集中的类

### DISCUSSION THREAD

**Discussion by TimonYang** [dotnetcore/Natasha#219] [Q&A]
大佬求教，
我在动态编译时需要加载其他程序集中的类，目前的代码是这样:
     
                 var action = NDelegate
                .RandomDomain()
                .ConfigClass(item => item
                .Name("ExecutePreprocessClass"))
                .Func<DataTable, object>(code);
 code中有对其他类的引用, 但这样是不行的，一直提示未能找到类型。
 我又尝试下面这种形式，但是并没有什么差别，还是一样的异常，这该如何是好？
           
            AssemblyCSharpBuilder builder = new AssemblyCSharpBuilder("ExecuteAssembly");
            builder.Domain = NatashaManagement.CreateDomain("ExecuteDomain");
            //builder.Domain.UsingRecorder.Using("DynamicModels");
            byte[] data = File.ReadAllBytes(@".\Dynamic.dll");

            using (Stream stream = new MemoryStream(data))
            {
                builder.Domain.LoadFromStream(stream);
            }
          
            string fullCode = “code”;
            builder.Add(fullCode);

            Assembly assembly = builder.GetAssembly();

顺便说一下，代码中加载的Dynamic.dll 也是用第二种方式生成的。

**Comment by NMSAzulX** [comment]
1. 查看 Natasha 关于插件编程的章节。文档和博客园有相关文章。建议使用API  LoadPlugin。
2. https://github.com/dotnetcore/Natasha/blob/main/src/Natasha.CSharp/Natasha.CSharp/MultiDomain/Component/Domain/NatashaReferenceDomain.cs#L68 这里是Core 版本以上的程序集加载处理，会自动添加using引用。 我猜测是插件加载的程序集中没有找到所需的类型。

**Reply by TimonYang** [reply]
感谢，是我犯了一个低级错误搞我心态，不应该用LoadFromStream，而该用Natasha中的LoadAssemblyFromStream，发完贴一会就发现了，所以自己给关了😂
